src/routes/Apps/App/TrainDialogs.tsx

```diff
@@ -102,51 +90,25 @@ function getColumns(intl: InjectedIntl): IRenderableColumn[] {
     let equalizeColumnWidth = window.innerWidth / 6
     return [
         {
-            key: 'firstInput',
-            name: Util.formatMessageId(intl, FM.TRAINDIALOGS_FIRSTINPUT),
-            fieldName: 'firstInput',
+            key: `tags`,
+            name: `Tags`,
+            fieldName: `Tags`,
             minWidth: 100,
             maxWidth: equalizeColumnWidth,
             isResizable: true,
             isSortedDescending: true,
-            render: trainDialog => {
-                let firstInput = getFirstInput(trainDialog);
-                if (firstInput) {
-                    return (<span className={textClassName(trainDialog)} data-testid="train-dialogs-first-input">
-                        {trainDialog.validity && trainDialog.validity !== CLM.Validity.VALID &&
-                            <OF.Icon
-                                className={`cl-icon ${ValidityUtils.validityColorClassName(trainDialog.validity)}`}
-                                iconName="IncidentTriangle"
-                            />
-                        }
-                        {firstInput}
-                    </span>)
-                }
-                return <OF.Icon iconName="Remove" className="notFoundIcon" />
-            },
-            getSortValue: trainDialog => {
-                let firstInput = getFirstInput(trainDialog)
-                return firstInput ? firstInput.toLowerCase() : ''
-            }
+            render: trainDialog => <TagsReadOnly className={textClassName(trainDialog)} tags={trainDialog.tags} />,
```

Comment:
@SDESkowronski 

These changes would break the tests. The `data-testid="train-dialogs-first-input"` which is used in `GetFirstInputs` and `ClickTraining`. I think changing `ClickTraining` to use updated `data-testid="train-dialogs-tags"` is fine since it doesn't matter which column we click, but the things like `EditTraining` depend on knowing firstInput which is no longer there.

For now I think the easiest solution is to add the `data-testid="train-dialogs-first-input"` and `data-testid="train-dialogs-last-input"` so the tests can still find and make use that information.

I see most of the tests rely on knowing the `firstInput` `lastInput` and `lastResponse` to interact with the train dialogs. Is there another way to know which row to click on?

I'm thinking in the future each train dialog created by the tests could add a specific tag or description such as `createdByTest` or `e2etest-branching`. Then search for that tag or description. This would allow finding specific dialogs to tests even if they have the same inputs and response fields.